pkgs/development/haskell-modules/configuration-nix.nix

```diff
@@ -680,7 +680,7 @@ self: super: builtins.intersectAttrs super {
   postgresql-pure = dontCheck super.postgresql-pure;
 
   retrie = overrideCabal super.retrie (drv: {
-    testToolDepends = [ pkgs.git pkgs.mercurial ];
+    testToolDepends = [ pkgs.buildPackages.git pkgs.buildPackages.mercurial ];
```

Comment:
I'm not very familiar with cross compilation stuff.

My understanding is that you want to get tools/libraries from `buildPackages` when they are used at build time on the build platform.  And you should get stuff from `pkgs` (or some variant of hostPkgsDrv or whatever) when you want to get tools/libraries that are used during runtime on the host platform.

In the case of cross-compilation, the build platform and the host platform will be different, so you have to be careful whether you get tools from `buildPackages` or `pkgs`.  But in the case of non-cross-compilation, you don't have to be careful whether you get things from `buildPackages` or `pkgs`, because they are the same.

------------------------------------------

Assuming the above is correct, my question here is as follows:

Aren't `testToolDepends` mostly binaries that are needed while running the tests?  Shouldn't they be binaries that match the host platform, since the test binaries are all going to be run on the host platform?

Reply:
> Aren't testToolDepends mostly binaries that are needed while running the tests? Shouldn't they be binaries that match the host platform, since the test binaries are all going to be run on the host platform?

The changes to the test tool dependencies were committed by accident. You are right philosophically: tests are to test what we are building and thus only have significance on the host platform, so anything tests need at runtime belongs to the host platform. What is necessary to build the tests would belong to the build platform.

However, as I explain in the commit message for f1dfc2a008ed92191bde5b76b045e5e41c6cb2c7, we don't need to care about this *at all*. We can't execute the tests anyways since we can't run the test binary on the build platform. Thus tests are always disabled when cross compiling:

> Note that this has intentionally not been for test tool dependencies:
> Like in `stdenv.mkDerivation` we need to view tests as being executed
> on the *host platform* which is why we can't run tests while cross
> compiling anyways. In practice this is not an important distinction,
> though: `pkgs.buildPackages` and `pkgs` are almost identical in the
> native case.

---

> In this case, I imagine that the test tool depends should be taken from the dynamically-linked glibc build platform, since that is bound to have more packages working (and a lot of these packages will probably already be in /nix/store).

In this case we also don't run test and it may be quite some effort to get them working at this point if we ever wanted to do that. Theoretically we could change the behavior of `stdenv.mkDerivation` to do a check if the build platform can execute binaries for the host platform to decide whether to execute tests, but this is not the case at the moment.